utils/local_remote_settings/__main__.py

```diff
@@ -0,0 +1,661 @@
+    if not vocab:
+        raise Exception("Could not find the vocab in artifacts")
+
+    model_path = models_path / f"{experiment_name}-{src}-{trg}-{export_task.status.taskId}"
+    model_path.mkdir(exist_ok=True)
+    config_path = model_path / "config.yml"
+
+    logger.info("Ensuring the artifacts are downloaded")
+
+    # Write out the config.
+    if config_path.exists():
+        logger.info(f"✅ {config_path}")
+    else:
+        config_text = yaml.safe_dump(config)
+        with config_path.open("wt") as config_file:
+            config_file.write(config_text)
+            logger.info(f"Destination: {config_path}")
+
+    downloads = [
+        (lex, "lex.s2t.bin"),
+        (model, "model.bin"),
+        (vocab, "vocab.spm"),
+    ]
+
+    # 1. Cache the artifact files locally to `model_path`
+    # 2. Add them as records to Remote Settings.
+    for artifact, filename in downloads:
+        # First cache it locally.
+        destination = model_path / filename
+        if destination.exists():
+            logger.info(f"✅ {destination}")
+        else:
+            zipped_file = f"{destination}.gz"
+            stream_download_to_file(
+                queue.buildUrl("getLatestArtifact", export_task.status.taskId, artifact.name),
+                zipped_file,
+            )
+            # Decompress the gzip.
+            logger.info(f"Decompressing from {zipped_file} to {destination}")
+            with gzip.open(zipped_file, "rb") as f_in:
+                with open(destination, "wb") as f_out:
+                    shutil.copyfileobj(f_in, f_out)
+
+        # Add it to Remote Settings, which retain it in-memory.
+        record = ModelRecord(
+            name=filename,
+            schema=0,
+            # Fake a locale out of the language tag and experiment name, e.g. "en-testDecoderSize"
+            fromLang=make_valid_locale_code(src, experiment_name),
+            toLang=make_valid_locale_code(trg, experiment_name),
+            version="1.0",
+            fileType=filename.split(".")[0],  # lex, model, vocab
+            attachment=None,
+            filter_expression="",
+            id=str(uuid4()),
+            last_modified=1728419357986,  # This is just a plausible static value.
+        )
+        remote_settings.create_record(
+            id=record.id,
+            collection=models_collection,
+            bucket=bucket,
+            data=json.loads(record.json()),
+        )
+        remote_settings.add_attachment(
+            id=record.id,
+            filepath=destination,
+            collection=models_collection,
+            bucket=bucket,
+        )
+        logger.info(f"Attachment added {record.name} {record.version}")
+
+
+# I9uKJEPvQd-1zeItJK0cOQ decoder-base
+# aAVZJcsXQg-vfGIjHmcCTw decoder-depth-3
+# e1DMdEzNSGyGhdjaWFYpxQ decoder-depth-6
+# Nu0YuyBLRuiimYaBKE0RcQ decoder-emb-biger
+# OABpAkBMQvapHE1lNozs0A decoder-ffn-bigger
+# TaeCdUs5Rqq7w1Tbf1PShQ decoder-tiny
```

Comment:
> question

Are these leftover debugging comments?

Reply:
Yeah, they were my IDs I was using for testing, thanks!